### PR TITLE
Fixes silicons from being unable to pick up cells from cell chargers

### LINKED ISSUE #22526: Cyborgs cannot use the magnetic gripper to remove power cells from cell chargers
**Issue Description**:
The magnetic gripper cannot remove power cells from cell chargers, despite the fact that it can be used to grab a cell and place it on the charger.

**What did you expect to happen**: 
For the magnetic gripper to be able to retrieve the cell from the charger.

**What happened instead**:
The gripper just uselessly bapped at the charger.

**Why is this bad/What are the consequences**:
Cyborgs cannot be helpful and charge loose power cells.

**Steps to reproduce the problem**:
Be a cyborg, pick up a power cell, place it on a cell charger, try to remove it.

**When did the problem start happening**:
Since the beginning of time.

**Extra information**:
No additional information.

### PR DESCRIPTION
**Pull request by meow20**
## What Does This PR Do
Fixes silicons from being unable to pick up cells from cell chargers

## Why It's Good For The Game
Its inconsistent, why can they insert them, but can't remove em?
fixes #22526

## Testing
Spawned in as a drone with a cell charger and few power cells, doing various combinations of adding cells, removing them, adding cells when cell was already inserted, dropping cells all resulted in correct behaviour.

## Changelog
:cl:
fix: Fixes silicons from being unable to pick up cells from cell chargers
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/drone/drone_items.dm [hunk 1]
```diff
@@ -137,6 +137,14 @@
 				A.update_icon()
 
 				user.visible_message("<span class='warning'>[user] removes the power cell from [A]!</span>", "You remove the power cell.")
+
+	else if (istype(target,/obj/machinery/cell_charger))
```
Comment (lewcc) [inline code/modules/mob/living/silicon/robot/drone/drone_items.dm:141]:
```suggestion
	else if (istype(target, /obj/machinery/cell_charger))
```

File: code/modules/mob/living/silicon/robot/drone/drone_items.dm [hunk 2]
```diff
@@ -137,6 +137,14 @@
...
+		var/obj/machinery/cell_charger/cell_charger = target
+		if(cell_charger.charging)
+			gripped_item = cell_charger.charging
+			cell_charger.removecell()
```
Comment (Contrabang) [inline code/modules/mob/living/silicon/robot/drone/drone_items.dm:145]: Some other stuff to be consistent
```suggestion
		if(cell_charger.charging)
			gripped_item = cell_charger.charging
			cell_charger.charging.add_fingerprint(user)
			cell_charger.charging.forceMove(src)
			cell_charger.removecell()
```

**Comment by Henri215** [comment]
image
Remove that empty line from the changelog

**Comment by meow20** [comment]
imagine being an admeme and not add fingerprints 💀

File: code/modules/mob/living/silicon/robot/drone/drone_items.dm [hunk 3]
```diff
@@ -137,6 +137,16 @@
 				A.update_icon()
 
 				user.visible_message("<span class='warning'>[user] removes the power cell from [A]!</span>", "You remove the power cell.")
+
+	else if (istype(target, /obj/machinery/cell_charger))
```
Comment (lewcc) [inline code/modules/mob/living/silicon/robot/drone/drone_items.dm:141]:
```suggestion
	else if(istype(target, /obj/machinery/cell_charger))
```

File: code/modules/mob/living/silicon/robot/drone/drone_items.dm [hunk 4]
```diff
@@ -137,6 +137,16 @@
...
+		var/obj/machinery/cell_charger/cell_charger = target
+		if(cell_charger.charging)
+			gripped_item = cell_charger.charging
+			cell_charger.charging.add_fingerprint(user)
+			cell_charger.charging.forceMove(src)
+			cell_charger.removecell()
+
+
```
Comment (lewcc) [inline code/modules/mob/living/silicon/robot/drone/drone_items.dm:149]:
```suggestion
```

**Review by lewcc — APPROVED** [review]

**Pull request merged**